op-deployer/pkg/deployer/integration_test/cli/command_test.go

```diff
@@ -0,0 +1,210 @@
+package cli
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/state"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInitCommandFlagValidation(t *testing.T) {
+	runner := NewCLITestRunner(t)
+
+	tests := []struct {
+		name        string
+		args        []string
+		expectError bool
+	}{
+		{
+			name:        "valid init command",
+			args:        []string{"init", "--l1-chain-id", "11155111", "--l2-chain-ids", "123"},
+			expectError: false,
+		},
+		{
+			name:        "missing required l2-chain-ids",
+			args:        []string{"init", "--l1-chain-id", "11155111"},
+			expectError: true,
+		},
+		{
+			name:        "invalid l1-chain-id format",
+			args:        []string{"init", "--l1-chain-id", "invalid", "--l2-chain-ids", "123"},
+			expectError: true,
+		},
+		{
+			name:        "invalid intent-type",
+			args:        []string{"init", "--l1-chain-id", "11155111", "--l2-chain-ids", "123", "--intent-type", "invalid"},
+			expectError: true,
+		},
+		{
+			name:        "valid intent-type custom",
+			args:        []string{"init", "--l1-chain-id", "11155111", "--l2-chain-ids", "123", "--intent-type", "custom"},
+			expectError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			workDir := runner.GetWorkDir()
+			args := append(tt.args, "--workdir", workDir)
+
+			if tt.expectError {
+				runner.ExpectErrorContains(t, args, nil, "")
```

Comment:
Are we able to pass error text to ensure we're triggering the expected error and not something else?